test/cache_common.c

```diff
@@ -4990,7 +4990,10 @@ resize_configs_are_equal(const H5C_auto_size_ctl_t *a, const H5C_auto_size_ctl_t
 {
     if (a->version != b->version)
         return (false);
-    else if (a->rpt_fcn != b->rpt_fcn)
+    /* Compare whether both report functions are enabled/disabled, not the exact pointer value
+     * (which can differ between DLL and executable on Windows)
+     */
+    else if ((a->rpt_fcn != NULL) != (b->rpt_fcn != NULL))
```

Comment:
Logic-wise this doesn't really seem like an appropriate replacement for determining if cache configurations are equal.